coral/src/app/features/topics/request/components/AdvancedConfiguration.tsx

```diff
@@ -0,0 +1,291 @@
+        <_Controller
+          name={name}
+          control={control}
+          render={({ field: { name, value } }) => {
+            return (
+              <>
+                <Editor
+                  data-testid="advancedConfiguration"
+                  height="300px"
+                  defaultLanguage={"json"}
+                  value={value}
+                  defaultValue={"{\n}"}
+                  onMount={handleEditorMount}
+                  onChange={(value) => {
+                    if (isString(value)) {
+                      setValue(name, value, {
+                        shouldValidate: true,
+                        shouldTouch: true,
+                        shouldDirty: true,
+                      });
+                    }
+                  }}
+                  onValidate={handleEditorValidation}
+                  options={{
+                    minimap: { enabled: false },
+                    folding: false,
+                    lineNumbers: "off",
+                    scrollbar: {
+                      vertical: "hidden",
+                      horizontal: "hidden",
+                      handleMouseWheel: false,
+                    },
+                  }}
+                />
+              </>
+            );
+          }}
+        />
+      </BorderBox>
+    </Box>
+  );
+}
+
+function setupAutocompleteAndTooltips(
+  monaco: ReturnType<typeof useMonaco>,
+  advancedConfigurationOptions: TopicAdvancedConfigurationOptions[] | undefined
+) {
+  if (monaco && advancedConfigurationOptions !== undefined) {
+    const completionProvider = monaco.languages.registerCompletionItemProvider(
+      "json",
+      {
+        provideCompletionItems: autocompleteOptionNames(
+          advancedConfigurationOptions
+        ),
+      }
+    );
+    const hoverProvider = monaco.languages.registerHoverProvider("json", {
+      provideHover: wordMatchesOptionName(advancedConfigurationOptions),
+    });
+
+    return () => {
+      completionProvider.dispose();
+      hoverProvider.dispose();
+    };
+  }
+}
+
+function getRange(model: editor.ITextModel, position: Position) {
+  const word = model.getWordUntilPosition(position);
+  return {
+    startLineNumber: position.lineNumber,
+    endLineNumber: position.lineNumber,
+    startColumn: word.startColumn,
+    endColumn: word.endColumn,
+  };
+}
+
+const autocompleteOptionNames = (
+  options: TopicAdvancedConfigurationOptions[]
+) => {
+  return (model: editor.ITextModel, position: Position) => {
+    const range = getRange(model, position);
+    return {
+      suggestions: options
+        .filter((option) => !model.getValue().includes(`"${option}"`))
+        .map((option) => {
+          return {
+            label: `"${option.name}"`,
+            range: range,
+            kind: 9, //languages.CompletionItemKind.Property
+            insertText: `"${option.name}": "",`,
+          };
+        }),
+    };
+  };
+};
+
+const wordMatchesOptionName = (
```

Comment:
The naming threw me off a bit, because the function sounds like it would return a boolean. Maybe something like "matchWordToOptionName"?